astroquery/splatalogue/core.py

```diff
@@ -76,14 +76,14 @@ def set_default_options(self, **kwargs):
         """
         self.data.update(self._parse_kwargs(**kwargs))
 
-    def get_species_ids(self, *, restr=None, reflags=0, recache=False):
+    def get_species_ids(self, regex_str=None, *, reflags=0, recache=False):
```

Comment:
and if rename happens, you may want to use a name that's more informative. Maybe `species_regex`?

Reply:
OK, I'm willing to abandon that.  `restr` wasn't great, it was just short.  It... kind of was never intended to be a `kwarg`, just an internal required positional variable.

Reply:
haha, good reminder that internal stuff should be named well, too. Or at least the stuff that ends up in signatures and docstrings.
I'm happy to push commits to this, but it will need to wait until at least the evening.

Reply:
I renamed the variable, but I still prefer to keep it as a positional argument.  I added the deprecation warning.